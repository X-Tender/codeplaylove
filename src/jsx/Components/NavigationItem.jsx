import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const propTypes = {
	children: PropTypes.node,
	to: PropTypes.string.isRequired,
	style: PropTypes.string.isRequired,
};

const defaultProps = {
	children: null,
};

const NavigationItem = ({ children, to, style }) => (
	<li className="navigation__list-item">
		<i className={`fas fa-${style}`} />
		<NavLink
			activeClassName="navigation__link--active"
			className={`navigation__link navigation__link--${style}`}
			exact
			to={to}
		>
			{children}
		</NavLink>
	</li>
);

NavigationItem.propTypes = propTypes;
NavigationItem.defaultProps = defaultProps;

export default NavigationItem;
