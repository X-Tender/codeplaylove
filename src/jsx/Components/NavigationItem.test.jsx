import React from 'react';
import { shallow } from 'enzyme';
import NavigationItem from './NavigationItem';

describe('NavigationItem', () => {
	const children = 'Home';
	const to = '/home';
	const style = 'house';

	const comp = shallow(
		<NavigationItem style={style} to={to}>
			{children}
		</NavigationItem>
	);

	it('renders correctly', () => {
		expect(comp).toMatchSnapshot();
	});

	it('renders the correct props', () => {
		expect(comp.find(`i.fas.fa-${style}`).exists()).toBe(true);
		expect(comp.find('NavLink').prop('to')).toEqual(to);
	});
});
