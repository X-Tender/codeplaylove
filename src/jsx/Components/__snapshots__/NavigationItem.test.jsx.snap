// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem renders correctly 1`] = `
<li
  className="navigation__list-item"
>
  <i
    className="fas fa-house"
  />
  <NavLink
    activeClassName="navigation__link--active"
    aria-current="page"
    className="navigation__link navigation__link--house"
    exact={true}
    to="/home"
  >
    Home
  </NavLink>
</li>
`;
